modules/map/hooks/useNotificationPermission.ts

```diff
@@ -1,99 +1,61 @@
 import messaging from '@react-native-firebase/messaging'
-import { useMutation, useQuery } from '@tanstack/react-query'
 import * as Device from 'expo-device'
-import { useCallback, useEffect, useState } from 'react'
-import { Platform } from 'react-native'
+import { useEffect, useState } from 'react'
 
-import { clientApi } from '@/modules/backend/client-api'
-import { devicesOptions } from '@/modules/backend/constants/queryOptions'
-import { MobileDevicePlatform } from '@/modules/backend/openapi-generated'
+import { useRegisterDevice } from '@/modules/map/hooks/useRegisterDevice'
 import { PermissionStatus } from '@/utils/types'
 
-type Options =
-  | {
-      autoAsk?: boolean
-      skipTokenQuery?: boolean
-    }
-  | undefined
+export const useNotificationPermission = (options?: { skipTokenRegistration?: boolean }) => {
+  const { skipTokenRegistration = false } = options ?? {}
 
-export const useNotificationPermission = ({ autoAsk, skipTokenQuery }: Options = {}) => {
   const [permissionStatus, setPermissionStatus] = useState<PermissionStatus>(
     PermissionStatus.UNDETERMINED,
   )
 
-  const { data, refetch } = useQuery(devicesOptions(skipTokenQuery))
-
-  const registerDeviceMutation = useMutation({
-    mutationFn: async (token: string) =>
-      // TODO investigate if/why this request gets resent in infinite loop sometimes
-      clientApi.mobileDevicesControllerInsertMobileDevice({
-        token,
-        platform: Platform.OS === 'ios' ? MobileDevicePlatform.Apple : MobileDevicePlatform.Android,
-      }),
-    onSuccess: async () => {
-      await refetch()
-    },
-  })
-
-  const checkAndRegisterToken = useCallback(async () => {
-    const token = await messaging().getToken()
-
-    if (token && data && !data.devices.some((device) => device.token === token)) {
-      registerDeviceMutation.mutate(token)
-    }
-  }, [data, registerDeviceMutation])
-
-  const checkPermission = useCallback(async () => {
-    const authStatus = await messaging().hasPermission()
-    const enabled =
-      authStatus === messaging.AuthorizationStatus.AUTHORIZED ||
-      authStatus === messaging.AuthorizationStatus.PROVISIONAL
+  const { registerDeviceIfNotExists } = useRegisterDevice(skipTokenRegistration)
 
-    if (enabled) {
-      setPermissionStatus(PermissionStatus.GRANTED)
+  useEffect(() => {
+    const checkStatus = async () => {
```

Comment:
aren't functions more readable when outside useEffect? what about adding it into useCallback and outside of useEffect for better readability

Reply:
Are they?